app/components/search.js

```diff
@@ -94,65 +85,48 @@ class Search extends React.Component {
         this.handleNextSearch()
       }
     })
-    this.focus()
-  }
-
-  componentWillUnmount = () => {
-    globalEmitter.removeListener('hideWindow', this.handleSaveQuery)
-    keyboard.unbind('search')
-  }
 
-  componentDidUpdate = () => {
-    if (this.props.value === '') {
-      this.focus()
+    // https://stackoverflow.com/questions/23123138/perform-debounce-in-react-js#answer-24679479
+    this.handleQueryChange = debounce(_ => this.props.handleQueryChange(this.state.searchQuery), 100)
```

Comment:
Debounce is handled by the plugin blocks. Both root and prefix have a debounce.

http://zazuapp.org/documentation/blocks/#root-script

Maybe giving this a default for those who don't set it?

Reply:
I gather you would like this to handle debounce 
(this is just a extract and refactor of the debounce you mentioned):
https://gist.github.com/afaur/ff126a0c2d016e3b7fa66f679012c599

When you say "giving this a default" do you mean: 
Set up a zazu config option for the amount of time to deflect successive fn calls?